runtime/js/11_workers.js

```diff
@@ -88,7 +91,7 @@
           cancelable: false,
           data,
         });
-        return;
+        throw new Error(`Received data could not be parsed: '${msgData}'.`);
```

Comment:
This is surprising... looks like during some refactors dispatch `messageerror` event got lost :/ if my memory serves right we should throw this error only if `messageerror` event wasn't prevented (ie. `preventDefault()` **hasn't been called** when handling the event).

Reply:
@bartlomieju I'm not sure I understand. what do you mean by `massageerror `event was prevented? the webWorker can prevent the `messageerror`? 
and what should happen if the messageerror was not prevented? the host should send the `messageerror `back to the webWorker?

Reply:
@yonbav I think just revert this change and focus the PR and tests on the behaviour when posting `undefined`.

Reply:
Done